ToneMapper/TMTonalFeature.m

```diff
@@ -107,21 +180,42 @@ - (IBAction)sliderValueChanged:(UISlider *)sender {
 }
 
 - (IBAction)toneOptionsValueChanged:(UISegmentedControl *)sender {
```

Comment:
These 2 switch/case blocks could be avoided by taking advantage of the builtin key-value coding. You could associate a target key-path for every button/segment during setup, and save that key-path in a designated property once the segment is selected. When the slider value changes you can simply update the value of the property pointed by that designated property.